### DISCUSSION TITLE
Map click listener

### DISCUSSION THREAD

**Discussion by Psijic** [osmdroid/osmdroid#1935] [Q&A]
Can't find how to set on map click listener properly.
I tried with `mapView.setOnClickListener{}` - nothing here.
Found inside
```
       mGestureDetector = new GestureDetector(context, new MapViewGestureDetectorListener());
        mGestureDetector.setOnDoubleTapListener(new MapViewDoubleClickListener());
```
but mGestureDetector is private. Hope, no layouts need to be added at least?

**Comment by Psijic** [comment]
Looks like there is no even basic class and I need to make it and add to layouts.
```
class OsmdroidMapEventsReceiver(
    val onSingleTap: (GeoPoint) -> Unit = {}, val onLongPress: (GeoPoint) -> Unit = {}
) : MapEventsReceiver {
    override fun singleTapConfirmedHelper(p: GeoPoint): Boolean {
        onSingleTap(p)
        return true
    }

    override fun longPressHelper(p: GeoPoint): Boolean {
        onLongPress(p)
        return true // Return true to consume the event, false to allow further handling
    }
}
```

**Reply by Psijic** [reply]
After adding the listener layer, my markers became unclickable. So, need to return false. But there 2 actions happen same time.

**Reply by LostRat** [reply]
I extend Marker class, or polyline or overlay and the like, when I want to capture user interactions. Don't know if this is this best way to do it. 

```
public class MyMarkerClass extends Marker
...
```

Then, for example:

```
 @Override
  public boolean onLongPress(MotionEvent e, MapView mapView) {
    if (DEBUG) Log.d(TAG, "LongPressed:" + pt);
    return super.onLongPress(e, mapView);
  }
```
